From efe8be6fad4944717c09ced56c406b65bd9006d2 Mon Sep 17 00:00:00 2001
From: Vignesh Viswanathan <quic_viswanat@quicinc.com>
Date: Tue, 30 Jul 2024 19:26:38 +0530
Subject: [PATCH] drivers: remoteproc: Add makefile for upd test driver

Add makefile for remoteproc upd test driver and minor changes to compile
the driver successfully.

Change-Id: I744a285d64cf5c6e173c15de1be95aabf09b79a4
Signed-off-by: Vignesh Viswanathan <quic_viswanat@quicinc.com>
---
 drivers/remoteproc/Makefile      | 1 +
 drivers/remoteproc/qcom_common.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/remoteproc/Makefile b/drivers/remoteproc/Makefile
index 9bd622c52233..b87717cd669e 100644
--- a/drivers/remoteproc/Makefile
+++ b/drivers/remoteproc/Makefile
@@ -30,6 +30,7 @@ obj-$(CONFIG_QCOM_Q6V5_MSS)		+= qcom_q6v5_mss.o
 obj-$(CONFIG_QCOM_Q6V5_PAS)		+= qcom_q6v5_pas.o
 obj-$(CONFIG_QCOM_Q6V5_WCSS)		+= qcom_q6v5_wcss.o
 obj-$(CONFIG_QCOM_Q6V5_WCSS_SEC)	+= qcom_q6v5_wcss_sec.o
+obj-$(CONFIG_QCOM_Q6V5_WCSS_SEC)	+= qcom_q6v5_upd.o
 obj-$(CONFIG_QCOM_SYSMON)		+= qcom_sysmon.o
 obj-$(CONFIG_QCOM_WCNSS_PIL)		+= qcom_wcnss_pil.o
 qcom_wcnss_pil-y			+= qcom_wcnss.o
diff --git a/drivers/remoteproc/qcom_common.h b/drivers/remoteproc/qcom_common.h
index 9f3fb11224aa..839fe34f94c4 100644
--- a/drivers/remoteproc/qcom_common.h
+++ b/drivers/remoteproc/qcom_common.h
@@ -77,4 +77,5 @@ static inline bool qcom_sysmon_shutdown_acked(struct qcom_sysmon *sysmon)
 #endif
 s8 qcom_get_pd_asid(struct device_node *node);
 
+int q6v5_userpd_copy_bootargs(struct rproc *rproc, void *data);
 #endif
-- 
2.34.1

